diff/parse.go

```diff
@@ -513,9 +514,22 @@ func (r *HunksReader) ReadHunk() (*Hunk, error) {
 			r.hunk.Section = section
 		} else {
 			// Read hunk body line.
+
+			// If the line starts with `---` and the next one with `+++` we're
+			// looking at a non-extended file header and need to abort.
+			if bytes.HasPrefix(line, []byte("---")) {
+				ok, err := nextLineHasPrefix(r.reader, []byte("+++"))
```

Comment:
would this fail with EOF when theres --- in the last line? (If that's even possible).

Reply:
Yes! Fixed in 5726eda